lib/rails_param/coercion/time_param.rb

```diff
@@ -0,0 +1,26 @@
+module RailsParam
+  module Param
+    class Coercion
+      class TimeParam
+        attr_reader :param, :options, :type
+
+        TIME_TYPES = [Date, DateTime, Time].freeze
+
+
+        def initialize(param:, options: nil, type: nil)
+          @param, @options, @type = param, options, type
+        end
+
+        def coerce
+          if TIME_TYPES.include? type
```

Comment:
I don't think we need to do this check here. We'll blow up spectacularly with the `parse` or `strptime` call if this isn't the correct type. We certainly don't want to return `nil` if it's called with an invalid `type`.